runtime/vm/callin.cpp

```diff
@@ -1341,6 +1344,12 @@ jitFillOSRBuffer(struct J9VMThread *currentThread, void *osrBlock)
 void JNICALL
 initializeAttachedThread(J9VMThread *currentThread, const char *name, j9object_t *group, UDATA daemon, J9VMThread *initializee)
 {
+#if defined(J9VM_OPT_CRIU_SUPPORT)
```

Comment:
Im wondering if I should just acquire the `vmThreadListMutex` to make this case impossibl

Reply:
Actually, this doesnt work. If there is a GC when we are running the java hook in single thread mode then it will deadlock when the GC thread attempts to acquire exclusive VMAccess

Reply:
In that case, the original code is similar to what we do for exclusive, so it's probably OK.